DEVELOPING.md

```diff
@@ -8,6 +8,16 @@ Before running through the release it's good to run the build and the tests loca
 also test-drive the commit in an existing Bazel workspace to sanity check functionality.
 
 #### Steps
+1. (Optional) modify the `./python/pip_install/tools/requirements.txt` file and run `./tools/update_pip_deps.sh` script:
+   1. Using system python:
```

Comment:
How does one know whether they should use the system Python or toolchain Python?

Is there a certain case the system python has to be used for? I'd rather not recommend it at all because it just sounds like a source of confusion -- we don't know what Python version it is, so different people might get different results, and then we churn the deps unknowingly.

Reply:
Good point, did not want a case where the deps may break the bazel setup (e.g. incorrect coverage deps are updated). However, once the PR is merged we will have a known to work setup and we can do `git checkout` upon failures to update the dependencies.